pkg/backend/httpstate/backend.go

```diff
@@ -732,24 +732,27 @@ func (b *cloudBackend) runEngineAction(
 		return nil, err
 	}
 
+	// The backend.SnapshotManager and backend.SnapshotPersister will keep track of any changes to
+	// the Snapshot (checkpoint file) in the HTTP backend.
 	persister := b.newSnapshotPersister(ctx, u.update, u.tokenSource)
-	manager := backend.NewSnapshotManager(persister, u.GetTarget().Snapshot)
+	snapshotManager := backend.NewSnapshotManager(persister, u.GetTarget().Snapshot)
```

Comment:
Ah yes. Originally I had thought I needed to introduce another channel, but opted to just reuse the existing one we use to send the display events.

I kept the refactoring in because IMHO it leaves the code in a slightly clearer state. But if you disagree, I'm happy to just revert the changes.

Reply:
Nope, it looks fine. I think the only thing I'd do is move the definitions of the event channels and the snapshot manager closer to their uses, but that's just a matter of preference. What you have looks good.